### ISSUE TITLE
Disabling autoconnect for specific networks

### ISSUE THREAD

**Issue by Komic** [thelounge/thelounge#2712]
The Lounge's current behaviour is to automatically join all networks when started. I'm often offline from certain networks on purpose, and would like to be able to have that choice. Please consider adding a "_Autoconnect at start_" checkbox in the network editing panel.

**Comment by xPaw** [comment]
What's the actual use case for wanting this?

**Comment by dgw** [comment]
Same as the use case for having networks saved in a traditional client that don't connect on start-up, I'd say. Convenience of connecting without devoting resources to maintaining the connection, or not wanting to be always online on that network.

**Comment by McInkay** [comment]
Maybe this could be set when we disconnect from a network without removing it, we set it as not-automatically joinable?

**Comment by xPaw** [comment]
I believe we discussed this before, if a user manually disconnects, then it would persist between server restarts, that sounds fine for me. Thus there's no need to add a separate checkbox.

**Comment by Komic** [comment]
@xPaw All my connect commands are automated (great work with that feature!), so basically, every time I have to restart The Lounge, I will join _every_ network/channel, even those I don't want to be seen on.

@McInkay I'd honestly prefer something toggleable but I know you guys hate those so your call, really. I'd prefer that to the current behaviour. :)

**Comment by McInkay** [comment]
I'm just trying to think of a way to solve what you want without having to add a toggle as that's not necessarily the best way to do things.

Is this for networks that you just want to connect to for specific events and then you disconnect? Or are there other use cases that I'm missing? Trying to figure out why my solution wouldn't be ideal for you.

**Comment by Komic** [comment]
> Is this for networks that you just want to connect to for specific events and then you disconnect?

For example, or networks I'm fine idling on once connected but _after_ deciding to /connect in the first place. Another concern would be unexpected shutdowns (so without disconnecting from networks first): after a restart, unexpected things could happen.
Your solution would make most people happy (including me, even partially!), how about going that way but letting users add a `"neverAutojoin": true,` in their .json to override it? No need for an ugly checkbox, and the vast majority of people would be happy with the new behaviour.

**Comment by McInkay** [comment]
> For example, or networks I'm fine idling on once connected but after deciding to /connect in the first place. 

Why would you not want this to auto-reconnect if you are currently connected to? If you aren't explicitly saying "I don't want to be connected to it" then surely reconnecting is fine?

> Another concern would be unexpected shutdowns (so without disconnecting from networks first): after a restart, unexpected things could happen.

What sort of things are you thinking could happen here?  It would just reconnect to everything it was currently connected to.

> Your solution would make most people happy (including me, even partially!), how about going that way but letting users add a "neverAutojoin": true, in their .json to override it? 

This is what we are trying to avoid. There should be a way to solve this without hardcoding some functionality into the config. I guess we could go this way (although I would probably invert the config to `autojoin` and default to true) but I'd rather if we didn't have to.

**Comment by xPaw** [comment]
> This is what we are trying to avoid. There should be a way to solve this without hardcoding some functionality into the config. I guess we could go this way (although I would probably invert the config to autojoin and default to true) but I'd rather if we didn't have to.

Yep, I still don't see the use case for doing so.

**Comment by Komic** [comment]
> What sort of things are you thinking could happen here?

People seeing joins from different nicks on different networks at the same time. I don't know, I might be too paranoid, in which case sorry, forget about the nitpicking, no autojoin if manually disconnected sounds great. :)

**Comment by astorije** [comment]
I am with @xPaw and @McInkay for the arguments they have presented.
Additionally, I think this goes against the "always on" character of The Lounge, and the recent efforts made to ensure server restarts do not affect the user experience, whom does not have to care about a server restart (which is true in general, not specific to The Lounge).

Having a way to prevent reconnects on random restarts makes for an unpredictable experience, which is not good UX in general. Like @xPaw mentioned, the current behavior seems to me to be the most predictable and seamless scenario: if I was connected before restart, I will be connected after restart; if I was not connected before restart, I will not be connected after restart.

> (assuming you guys go down that road: the "don't auto-connect if manually disconnected" behaviour)

@xPaw, correct me if I'm wrong, but isn't that the behavior on `master`?

I will close this issue as all maintainers agree on not adding an extra new way to not reconnect after restart, but if my question above is a no, then I think we all agree it should behave that way.

**Comment by xPaw** [comment]
@astorije, it isn't, manual disconnection is not stored, so restarting the server will connect again.

**Comment by Komic** [comment]
Sorry if I wasn't clear but yeah, afaik there's _no_ way to prevent being autoconnected, I don't care about the specifics, this was just about implementing _a_ way.

**Comment by xPaw** [comment]
#2718

**Comment by astorije** [comment]
My bad, the way this was discussed made me think it was already in there. Thanks @xPaw!

**Comment by Dreytac** [comment]
I know this is an old issue but I don't know if it's worth opening a new issue. I just stumbled onto this problem and I have a use case for it.

I have *sketchy* internet at times which drops in and out every few minutes for a couple of hours (state of the art Australian NBN for ya...). I've been banned repeatedly from a couple of networks for bouncing in and out of the server. My thought to fix this would be to disable auto-reconnect on the affected servers like I used to with other IRC clients. If an option to disable auto-reconnect is not acceptable, is there another option be that would allow me to keep using The Lounge (and no, better internet is not an option :P)?
